x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/RestDelegatePkiAuthenticationAction.java

```diff
@@ -50,8 +50,8 @@ public List<Route> routes() {
     }
 
     @Override
-    protected Exception checkFeatureAvailable(RestRequest request) {
-        Exception failedFeature = super.checkFeatureAvailable(request);
+    protected Exception checkFeatureAvailable() {
+        Exception failedFeature = super.checkFeatureAvailable();
```

Comment:
All overriden method needs to call `super.checkFeatureAvailable()`. This to me suggests a chance to split this method in to two, e.g.:

```java
class SecurityBaseRestHandler {
    protected final checkFeatureAavailable() {
        // existing basic check
        doCheckFeatureAavailable();
    }

    protected abstract doCheckFeatureAavailable();
}
```

So that sub-classes just override `doCheckFeatureAavailable` without worrying about the super method.

Reply:
good thinking - I named this `innerCheckFeatureAvailable` so it uses the same convention like `innerPrepareRequest`.
also I could not make it abstract since not all subclasses implement it.